workspaces/mcp-chat-client/plugins/mcp-chat-client-backend/src/router.ts

```diff
@@ -0,0 +1,304 @@
+        headers = Object.fromEntries(
+          Object.entries(headersConfig as Record<string, any>).map(
+            ([key, value]) => [key, String(value)],
+          ),
+        );
+      }
+    }
+
+    let env: Record<string, string> | undefined;
+    if (serverConfig.has('env')) {
+      const envConfig = serverConfig.getConfig('env').get();
+      if (envConfig && typeof envConfig === 'object') {
+        env = Object.fromEntries(
+          Object.entries(envConfig as Record<string, any>).map(
+            ([key, value]) => [key, String(value)],
+          ),
+        );
+      }
+    }
+
+    return {
+      id: serverConfig.getOptionalString('id'),
+      name: serverConfig.getString('name'),
+      scriptPath: serverConfig.getOptionalString('scriptPath'),
+      npxCommand: serverConfig.getOptionalString('npxCommand'),
+      args: serverConfig.getOptionalStringArray('args'),
+      env,
+      url: serverConfig.getOptionalString('url'),
+      headers,
+      type: serverConfig.getOptionalString('type') as
+        | 'stdio'
+        | 'sse'
+        | 'streamable-http'
+        | undefined,
+    };
+  });
+}
+
+export async function createRouter({
+  mcpClientService,
+  config,
+}: {
+  httpAuth: HttpAuthService;
+  mcpClientService: MCPClientService;
+  config: RootConfigService;
+}): Promise<express.Router> {
+  const router = Router();
+  router.use(express.json());
+
+  // Configuration status endpoint
+  router.get('/config/status', async (_req, res) => {
+    try {
+      const providerStatus = mcpClientService.getProviderStatus();
+      const mcpServers =
+        config.getOptionalConfigArray('mcpChat.mcpServers') || [];
+      const serverList = mcpServers.map(serverConfig => ({
+        id: serverConfig.getOptionalString('id'),
+        name: serverConfig.getString('name'),
+        type: serverConfig.getOptionalString('type') || 'stdio',
+        hasUrl: serverConfig.has('url'),
+        hasNpxCommand: serverConfig.has('npxCommand'),
+        hasScriptPath: serverConfig.has('scriptPath'),
+      }));
+
+      res.json({
+        provider: providerStatus,
+        mcpServers: serverList,
+      });
+    } catch (error) {
+      res.status(500).json({
+        error: 'Failed to get configuration status',
+        details: error instanceof Error ? error.message : 'Unknown error',
+      });
+    }
+  });
+
+  // MCP Chat route
+  router.post('/chat', async (req, res) => {
+    try {
+      const { messages, enabledTools = [] } = req.body;
+
+      // Validate messages array
+      if (!messages || !Array.isArray(messages) || messages.length === 0) {
+        return res.status(400).json({ error: 'No query provided' });
+      }
+
+      const userQuery = messages[messages.length - 1]?.content;
+
+      if (!userQuery) {
+        return res.status(400).json({ error: 'No query provided' });
+      }
+
+      const serverConfigs = loadServerConfigs(config);
+      await mcpClientService.initMCP(serverConfigs);
```

Comment:
shouldn't these two steps be invoked at startup instead?

Reply:
makes sense 👍